### ISSUE TITLE
Hatsune Miku: Project Diva F [BLJM-60527]: trying to open non existent files, wrong encoding?

### ISSUE THREAD

**Issue by GhettoGirl** [RPCS3/rpcs3#1821]
The game 初音ミク -Project DIVA- F is trying to open files which are clearly not exists
```{PPU[0x110] Thread (200baad0) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/rom/rob/mot_MI!!!!!!') failed: not found```

There are several files starting with ```mot_MI``` in this directory
```
mot_MIK.farc
mot_MINIGAME01.farc
mot_MINIGAME02.farc
```
From the number of exclamation mark I would guess the game is trying to open one of the MINIGAME files. Could this be a bug in the filesystem code or comes this from somewhere else?

**LLE modules loaded to even enter the loading screen**
```yml
    - libavcdec.sprx
    - libdivx311dec.sprx
    - libdivxdec.sprx
    - libdmux.sprx
    - libdmuxpamf.sprx
    - libmvcdec.sprx
    - libpamf.sprx
    - librtc.sprx
    - libsjvtd.sprx
    - libsmvd2.sprx
    - libsmvd4.sprx
    - libsre.sprx
    - libsvc1d.sprx
    - libvdec.sprx
```
---

[Game Log](https://justpaste.it/vlir)
[RPCS3 Log / SPUJIT Log](https://mega.nz/#F!WxJVhBTQ!xnQfWhzkgGN7oNiQMzojTA)

**Comment by GhettoGirl** [comment]
I created a symlink from ```mot_MI!!!!!!``` to ```mot_MIK.farc```, it seems to be the right file. Now it stops with this.
This looks like an encoding bug to me :/
Does the PS3 use UTF-8 or something else, I'm curious.
Or something isn't right with my game dump. Btw I own this game and dumped it from the blu-ray disc.
All filenames are in UTF-8 on my filesystem.

```
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/rom/light_param/light_aetback.tx') failed: not found
{rsx::thread} RSX: draw_calls: 86, draw_call_setup: 337us, vertex_upload_time: 1104us, draw_call_execution: 1348us
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/rom/light_param/fog_aetback.txt') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/rom/light_param/glow_aetback.txt') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/rom/light_param/wind_aetback.txt') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIrom/objset/mikitm.farc') failed: not found
{rsx::thread} RSX: draw_calls: 86, draw_call_setup: 317us, vertex_upload_time: 993us, draw_call_execution: 1178us
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIrom/objset/mikitm.farc') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIrom/objset/mikitm.farc') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIrom/objset/mikitm.farc') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDI/rom/objset/mikitm.farc') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/rom/rob/mot_CMN.') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIrom/rob/mot_MI') failed: not found
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/./rom/mothead/mothead_MIK.bin') failed: not found
```

**Comment by paulsapps** [comment]
That looks like its possible the string getting corrupted since I assume its concatenated them to make that path. If its in a HLE module valgrind should find it.

**Comment by GhettoGirl** [comment]
The problem solved itself by running ```echo 1 > /proc/sys/vm/drop_caches```. It helps in most cases to get rid of old cached memory which conflicts between RPCS3 sessions. The output from above went away.
But there is something wrong with my dump, because now there are actually missing files. I need to redump the game to see if I can get those files and than hopefully get past the loading screen.

You were right with the memory corruption btw.

Also: The memory corrups pretty fast. I actually need to drop caches every time before starting up RPCS3 again or a lot of strage things happen, like SEGFAULT and game crashes, etc. :confused:

---

**Note to developers:** On Linux you need to write a proper clean up code sometime in the future and handle all this signals:
 - SIGINT
 - SIGTERM
 - SIGSEGV (very important)
 - SIGABRT

to avoid the stuff mentioned above.

**Comment by Nekotekina** [comment]
Try SPU ASMJIT, interpreter contains some bugs when compiled with clang or gcc.

**Comment by GhettoGirl** [comment]
With SPU ASMJIT atleast the memory doesn't corrupt that fast, but sometimes I still need to drop caches.

I redumped my game disc, but I can't get past loading screen, there are missing files and they are not on my game disc :confused: 
```
PS3_GAME/USRDIR/rom/light_param/fog_aetback.txt
PS3_GAME/USRDIR/rom/light_param/glow_aetback.txt
PS3_GAME/USRDIR/rom/light_param/wind_aetback.txt
PS3_GAME/USRDIR/./rom/mothead/mothead_MIK.bin
```

On a real PS3 the game boots normally, can it be that my drive can't read some files?
I'm using a **LG BD-RE WH16NS40** as of seen on **[this](http://www.emunewz.net/forum/showthread.php?tid=173113&pid=231047#pid231047)** list.

I don't have a jailbroken PS3, I have a PS3-4000 Super Slim which isn't jailbreakable so I can't dump the game from there to compare the rips.

**Comment by Nekotekina** [comment]
Ah wait, are those files decrypted correctly? I think it's possible to google their hashes to verify.

Regarding files which are not found, it's normal to see those error messages because if they don't normally exist they are just not supposed to be accessed.

**Comment by GhettoGirl** [comment]
> Ah wait, are those files decrypted correctly? I think it's possible to google their hashes to verify.

I don't even see them on the game disc when browsing it. Their are simple not existent :(
I guess the game needs this files, which is the reason why it freezes in middle of the loading screen.

This is the last message in the output
```
{PPU[0xf] Thread (main_thread) [0x009c6790]} sys_fs: sys_fs_stat('/dev_bdvd/PS3_GAME/USRDIR/./rom/mothead/mothead_MIK.bin') failed: not found
```
were the loading screen begins to hang.
It is also not on the US version [BLUS-31319] of the game.

**Comment by Nekotekina** [comment]
No, I mean all other files which are extracted.

**Comment by GhettoGirl** [comment]
> No, I mean all other files which are extracted.

Oh I see. I googled some file hashes - no results. But I guess that they are probably decrypted, since I can view every image and watch any movie just fine without problems. Also all the plain text files are readable and UTF-8 encoded.

**Comment by Nezarn** [comment]
Those files are missing from PSN version too, maybe game devs just forgot to remove them? (and also PSN version works just fine)

Idk how much the disc and PSN version differs, but here is the file list (with removed movies) with crc:

https://www.dropbox.com/s/z9oh9vkjg2dhrsf/NPUB31241.sfv?dl=0

BTW on which loading screen do you get stuck? Some songs have movies and they will cause infinite loading screen (for me even with LLE modules movies doesn't work, so i removed them for workaround -> then game is playable without any LLE)

**Comment by GhettoGirl** [comment]
I don't think the problem is because of the missing files now. I created some symlink to trick the game that this files exists.
```
PS3_GAME/USRDIR/rom/light_param/fog_aetback.txt  -> fog_tst.txt
PS3_GAME/USRDIR/rom/light_param/glow_aetback.txt -> glow_tst.txt
PS3_GAME/USRDIR/rom/light_param/wind_aetback.txt -> wind_tst.txt
PS3_GAME/USRDIR/rom/mothead/mothead_MIK.bin      -> mothead_CMN.bin
```

But it still hangs in middle of the loading screen.

@Nezarn It hangs at the loading screen after the main menu when selecting "Loading Game", after 5 seconds.
Here are the logs: https://mega.nz/#F!yth3hA6a!RoOQYHDBnOGYfX7h_WVq-Q

**Comment by GhettoGirl** [comment]
> then game is playable without any LLE

When I disable all LLE modules mentioned in my report
```
{PPU[0x1ee] Thread (Demuxer[0x1ed] Thread)} N3fmt9exceptionE thrown: operator()(): dmuxEnableEs: unknown filter (0xe0, 0x0, 0x0, 0x0)
(in file /build/rpcs3/rpcs3/Emu/Cell/Modules/cellDmux.cpp:678)
```
Has this something todo with movies?

**Comment by Nezarn** [comment]
@GhettoGirl yes. (since the "codec" the movie has isn't implemented yet)

**Comment by GhettoGirl** [comment]
UPDATE: I've removed the symlink I created before, now it doesn't stuck anymore, but it infinitely loads.

EDIT: It stucks again after clearing the cache ```echo 1 > /proc/sys/vm/drop_caches``` :confused:

**Comment by Nezarn** [comment]
@GhettoGirl do you have linux only? I think this is an OpenGL regression, since even in windows it gets stuck with opengl exactly where your game gets stuck

![rpcs3_2016-06-26_00-28-39](https://cloud.githubusercontent.com/assets/3817985/16359429/0222d8be-3b35-11e6-8b05-ed36c8dd914c.png)

@DHrpcs3 (i hope im "pinging" the correct person)
after stopping the emulation:

`·F {rsx::thread} class gl::glsl::compilation_exception thrown: compilation failed: '0(88) : error C1008: undefined variable "fogc"`

log: https://gist.github.com/Nezarn/b5d99d96e7728535dd12e3c60c45ed08
ogl cache: https://www.dropbox.com/s/0v9cc74fceyaxym/cache.zip?dl=0

**Comment by GhettoGirl** [comment]
@Nezarn I have Linux as the only host OS, and tons of other operating systems as virtual machines for different use cases. OpenGL performance in VMs is bad/not ideal.
I don't think running RPCS3 in my Win10 Pro VM would be a good idea.

**Comment by GhettoGirl** [comment]
The latest update is able to get past loading screen. There are a lot of other problems now. I'm opening another issue for this with all the details and logs.

Regards
